sylvia-derive/src/message.rs

```diff
@@ -303,7 +303,7 @@ impl<'a> EnumMessage<'a> {
                     #[allow(clippy::derive_partial_eq_without_eq)]
                     #[derive(#sylvia ::serde::Serialize, #sylvia ::serde::Deserialize, Clone, Debug, PartialEq, #sylvia ::schemars::JsonSchema, cosmwasm_schema::QueryResponses)]
                     #[serde(rename_all="snake_case")]
-                    pub enum #unique_enum_name #generics #where_clause {
```

Comment:
I removed it because of duplicated bound on `Deserialize` impl.

Without the where clause
```rust
#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
pub enum MyEnum<Msg> {}
```

We will get no additional bounds here
```rust
    impl<'de, Msg> _serde::Deserialize<'de> for MyEnum<Msg>
    where
        Msg: _serde::Deserialize<'de>,
    {
```

But if user will provide a `Deserialize` bound here

```rust
#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
pub enum MyEnum<Msg>
where
    for<'a> Msg: Deserialize<'a>,
```

`Deserialize` will expand to

```rust
    impl<'de, Msg> _serde::Deserialize<'de> for MyEnum<Msg>
    where
        for<'a> Msg: Deserialize<'a>,
        Msg: _serde::Deserialize<'de>,
```

This also aligns with what `dtolnay` said here https://github.com/serde-rs/json/issues/469#issuecomment-408620000. Basically to keep `behavior` bounds out of the type and move them to `impl block`.

Reply:
Ok, it makes sense. First time I've seen I didn't yet know you moved it. I will check generated code today.